### PR TITLE
docs(third-party-commitizen): add commitizen-deno-provider

### PR DESCRIPTION
**Pull request by dhmmasson**
## Description
Add a new third party commitizen: a Deno provider. 

## Checklist

- [-] Add test cases to all the changes you introduce
- [-] Run `./scripts/format` and `./scripts/test` locally to ensure this change passes linter check and test
- [-] Test the changes on the local machine manually
- [x] Update the documentation for the changes

## Additional context
Related to #1257

### CODE REVIEW

File: docs/third-party-commitizen.md [hunk 1]
```diff
@@ -64,23 +65,50 @@ pip install cz-conventional-gitmoji
...
+
+A provider for Deno projects. The provider updates the version in deno.json, deno.lock and jsr.json files.
+
+#### Installation
+
+
+```
+pip install commitizen-deno-provider
+```
+
```
Comment (Lee-W) [inline docs/third-party-commitizen.md:98]:
```suggestion

```sh
pip install commitizen-deno-provider
```

```

File: docs/third-party-commitizen.md [hunk 2]
```diff
@@ -64,23 +65,50 @@ pip install cz-conventional-gitmoji
...
+#### Usage
+
+Add `deno-provider` to your configuration file.
+
+Example for `.cz.yaml`:
+
+```yml
```
Comment (Lee-W) [inline docs/third-party-commitizen.md:105]:
```suggestion
```yaml
```

let's keep it this way for consistency

**Comment by dhmmasson** [comment]
I have updated as requested my section [00d5ab8](https://github.com/commitizen-tools/commitizen/pull/1259/commits/00d5ab89816fafd735d85d2a93906cfcf7c97631)
I have also proposed a more general overall in the general direction of the comments : [e7045a8](https://github.com/commitizen-tools/commitizen/pull/1259/commits/e7045a842e52a6d84a6f790f18bcae2494ec4650)
- uniform use of sh/bash in the code blocks
- More uniform level heading (top level for provider, installation/usage for all templates)

I can revert the last commit if the change is an issue

**Review by Lee-W — APPROVED** [review]

**Comment by Lee-W** [comment]
> I have updated as requested my section [00d5ab8](https://github.com/commitizen-tools/commitizen/pull/1259/commits/00d5ab89816fafd735d85d2a93906cfcf7c97631) I have also proposed a more general overall in the general direction of the comments : [e7045a8](https://github.com/commitizen-tools/commitizen/pull/1259/commits/e7045a842e52a6d84a6f790f18bcae2494ec4650)
> 
>     * uniform use of sh/bash in the code blocks
> 
>     * More uniform level heading (top level for provider, installation/usage for all templates)
> 
> 
> I can revert the last commit if the change is an issue, one thing I didn't know how to do with mkdocs is to limit the TOC to level 3, as it is a bit to busy right now

Thanks sooo much! You have already done so many great things!

**Pull request merged**